src/sessions.ts

```diff
@@ -286,12 +289,12 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
    */
   startTransaction(options?: TransactionOptions): void {
     if (this[kSnapshotEnabled]) {
-      throw new MongoDriverError('Transactions are not allowed with snapshot sessions');
+      throw new MongoCompatibilityError('Transactions are not allowed with snapshot sessions');
     }
 
     assertAlive(this);
     if (this.inTransaction()) {
-      throw new MongoDriverError('Transaction already in progress');
+      throw new MongoAPIError('Transaction already in progress');
```

Comment:
Since we are moving MongoTransactionError to be child of MongoAPIError, should we change this to MongoTransactionError and just go ahead and move MongoTransactionError to be a child of MongoAPIError in this PR?

Reply:
If not, we should also mark this for replacement in NODE-3485